frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install.tsx

```diff
@@ -31,34 +39,34 @@ export const CreateAttachedDevicesCluster: React.FC<CreateAttachedDevicesCluster
 
   const { appName, ns } = match.params;
   const [hasNoProvSC, setHasNoProvSC] = React.useState(false);
-  const [isLsoPresent, setIsLsoPresent] = React.useState(false);
-  const [allDataLoaded, setAllDataLoaded] = React.useState(false);
-  const [lsoNs, setLsoNs] = React.useState('');
+  const [lsoSubscription, setLsoSubscription] = React.useState<K8sResourceKind>();
+  const isDataLoaded = React.useRef(false);
+  const lsoNs = getNamespace(lsoSubscription);
+  const lsoVersion = getOperatorVersion(lsoSubscription);
+
+  const subscriptionResource: WatchK8sResource = {
+    kind: referenceForModel(SubscriptionModel),
+    isList: true,
+  };
   const [subscription, subscriptionLoaded, subscriptionLoadError] = useK8sWatchResource<
     K8sResourceKind[]
-  >(LSOSubscriptionResource);
+  >(subscriptionResource);
+
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    name: lsoVersion,
+    namespaced: true,
+    namespace: lsoNs,
+    isList: false,
+  };
+  const [csv, csvLoaded, csvLoadError] = useK8sWatchResource<K8sResourceKind>(csvResource);
+
+  if (subscriptionLoaded && csvLoaded) isDataLoaded.current = true;
+  const isLsoCsvSucceeded = (lsoNs && lsoVersion && csv?.status?.phase === 'Succeeded') || false;
 
   React.useEffect(() => {
-    if (subscriptionLoadError || (!subscription.length && subscriptionLoaded)) {
-      setIsLsoPresent(false);
-      setAllDataLoaded(true);
-    } else if (subscriptionLoaded && !_.isEmpty(subscription[0])) {
-      fetchK8s(
-        ClusterServiceVersionModel,
-        subscription[0]?.status?.installedCSV,
-        subscription[0]?.metadata?.namespace,
-      )
-        .then(() => {
-          setIsLsoPresent(true);
-          setLsoNs(subscription[0]?.metadata?.namespace);
-          setAllDataLoaded(true);
-        })
-        .catch(() => {
-          setIsLsoPresent(false);
-          setAllDataLoaded(true);
-        });
-    }
-  }, [subscription, subscriptionLoaded, subscriptionLoadError]);
+    setLsoSubscription(subscription.find((item) => item?.spec?.name === LSO_OPERATOR));
+  }, [subscription]);
```

Comment:
This will lower the renders
```suggestion
React.useEffect(() => {
    const subscription = subscriptions.find((item) => item?.spec?.name === LSO_OPERATOR);
    if (lsoSubscription?.spec?.name !== LSO_OPERATOR) setLsoSubscription(subscription);
  }, [lsoSubscription, subscriptions]);
```